src/utils/anonymousUserMerge.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  SHARED_PREF_EMAIL,
+  SHARED_PREF_USER_ID,
+  ENDPOINT_GET_USER_BY_USERID,
+  ENDPOINT_GET_USER_BY_EMAIL,
+  ENDPOINT_MERGE_USER
+} from 'src/constants';
+import { AnonymousUserEventManager } from './anonymousUserEventManager';
+import { baseIterableRequest } from '../request';
+import { IterableResponse } from '../types';
+
+export class AnonymousUserMerge {
+  private anonymousUserManager = new AnonymousUserEventManager();
+
+  mergeUserUsingUserId(destinationUserId: string): void {
```

Comment:
mergeUserUsingUserId and mergeUserUsingEmail should be joined. we can determine whether to use one or the other in the method itself.

Reply:
We need both methods, as we need to call separate APIs for each and it will make method complex to put many checks.

Reply:
Here is an [example](https://github.com/Iterable/iterable-web-sdk/pull/304) of how this could be cleaned up. This hasn't been tested yet and the unit tests haven't been changed but should give you an idea.

Also, we should consider returning something from that call as opposed to `void`. I imagine devs would want to know whether it succeeded or failed.

Reply:
Another thought, we should also pass in the `source`, so the `mergeUser` function takes in `destination` and `user` so we don't have that weird early return statement. That, way, the method is only concerned about making the proper api calls, not pulling data from local storage. If there's no local storage data or the target equals the source, then we shouldn't call the method to begin with. Those checks should happen higher up in the call order.

Reply:
So really, the `mergeUser` method should take in this type as it's param, same as the `callMergeApi` in my example and the logic to get those params should happen outside the scope of this method somewhere. 

```
  type MergeApiParams = {
  sourceEmail: string | null;
  sourceUserId: string | null;
  destinationEmail: string | null;
  destinationUserId: string | null;
}
```

Reply:
Thank you for the new vision for the development. I have implemented that way and also modified the test cases accordingly.